### PR TITLE
Support Rxdart 0.22.0

### PR DESCRIPTION
**Pull request by yprihodko**

### CODE REVIEW

File: lib/hydrated.dart [hunk 1]
```diff
@@ -45,15 +45,22 @@ class HydratedSubject<T> extends Subject<T> implements ValueStream<T> {
   String Function(T value) _persist;
   void Function() _onHydrate;
 
+  final Future<SharedPreferences> _prefsFutureOverride;
+
+  Future<SharedPreferences> get _prefsFuture async {
```
Comment (mateusfccp) [inline lib/hydrated.dart:50]: `async` is not necessary here, as you are not using `await` notation.

File: lib/hydrated.dart [hunk 2]
```diff
@@ -93,13 +99,13 @@ class HydratedSubject<T> extends Subject<T> implements ValueStream<T> {
         persist,
         onHydrate,
         controller,
-        Rx.defer<T>(
+        new Observable<T>.defer(
```
Comment (mateusfccp) [inline lib/hydrated.dart:102]: `new` is not needed and discouraged by Effective Dart.

File: lib/hydrated.dart [hunk 3]
```diff
@@ -93,13 +99,13 @@ class HydratedSubject<T> extends Subject<T> implements ValueStream<T> {
...
             () => wrapper.latestValue == null
                 ? controller.stream
-                : controller.stream
-                    .startWith(wrapper.latestValue),
+                : new Observable<T>(controller.stream).startWith(wrapper.latestValue),
```
Comment (mateusfccp) [inline lib/hydrated.dart:105]: `new` is not needed and discouraged by Effective Dart.

File: pubspec.yaml [hunk 4]
```diff
@@ -4,14 +4,14 @@ version: 1.2.5+2
...
-  sdk: ">=2.6.0 <3.0.0"
+  sdk: ">=2.5.0 <3.0.0"
 
 dependencies:
   flutter:
     sdk: flutter
 
-  rxdart: ^0.24.1
-  shared_preferences: ^0.5.7+3
+  rxdart: ^0.22.0
```
Comment (mateusfccp) [inline pubspec.yaml:13]: Why are you reverting `rxdart` version? It was me myself that requested and made the PR to upgrade it to the latest version.

People using the latest `rxdart` won't be able to use the package.

Comment (mateusfccp) [reply pubspec.yaml:13]: Refer to #14.

Comment (yprihodko) [reply pubspec.yaml:13]: This branch is not meant to be merged into master and released. I'll delete it shortly.
We were using it for an internal project that required rxdart 0.22.0.

Comment (mateusfccp) [reply pubspec.yaml:13]: I see, sorry.

**Pull request merged**